chimney/src/main/scala/io/scalaland/chimney/TransformerFSupport.scala

```diff
@@ -0,0 +1,154 @@
+    * between collections.
+    *
+    * @param it  iterator of elements of type `A`
+    * @param f   function to apply to elements of type `A`, returning `F[B]`
+    * @param fac factory for collection type `M`
+    * @tparam M  type of collection where transformed elements are stored; note that this is not
+    *            a type constructor, but a type with applied argument, so it can be List[B], Map[K, V], etc.
+    * @tparam A  type of elements being iterated
+    * @tparam B  target element type of function `f`
+    * @return wrapped collection of type `F[M]`
+    */
+  def traverse[M, A, B](it: Iterator[A], f: A => F[B])(
+      implicit fac: Factory[B, M]
+  ): F[M]
+}
+
+object TransformerFSupport {
+
+  /** `TransformerFSupport` instance for `Option`
+    */
+  implicit def TransformerFOptionSupport: TransformerFSupport[Option] =
+    new TransformerFSupport[Option] {
+
+      def pure[A](value: A): Option[A] = Option(value)
+
+      def product[A, B](fa: Option[A], fb: => Option[B]): Option[(A, B)] = {
+        fa.flatMap(a => fb.map(b => (a, b)))
+      }
+
+      def map[A, B](fa: Option[A], f: A => B): Option[B] = {
+        fa.map(f)
+      }
+
+      def traverse[M, A, B](it: Iterator[A], f: A => Option[B])(
+          implicit fac: Factory[B, M]
+      ): Option[M] = {
+        val b = fac.newBuilder
+        var wasNone = false
+        while (!wasNone && it.hasNext) {
+          f(it.next()) match {
+            case None     => wasNone = true
+            case Some(fb) => b += fb
+          }
+        }
+        if (wasNone) None else Some(b.result())
+      }
+    }
+
+  /** `TransformerFSupport` instance for `Either[C[E], +*]`.
+    *
+    * @param ef factory for error accumulator collection
+    * @tparam E error type
+    * @tparam C error accumulator type constructor
+    */
+  implicit def TransformerFEitherErrorAccumulatingSupport[E, C[X] <: IterableOnce[X]](
+      implicit ef: Factory[E, C[E]]
+  ): TransformerFSupport[Either[C[E], +*]] = new TransformerFSupport[Either[C[E], +*]] {
+
+    def pure[A](value: A): Either[C[E], A] = Right(value)
+
+    def product[A, B](fa: Either[C[E], A], fb: => Either[C[E], B]): Either[C[E], (A, B)] = {
+      (fa, fb) match {
+        case (Right(ok1), Right(ok2)) =>
+          Right((ok1, ok2))
+        case (left1, left2) =>
+          val eb = ef.newBuilder
+          left1.left.foreach(eb ++= _)
+          left2.left.foreach(eb ++= _)
+          Left(eb.result())
+      }
+    }
+
+    def map[A, B](fa: Either[C[E], A], f: A => B): Either[C[E], B] = {
+      fa match {
+        case Right(a)   => Right(f(a))
+        case Left(errs) => Left(errs)
+      }
+    }
+
+    def traverse[M, A, B](it: Iterator[A], f: A => Either[C[E], B])(
+        implicit fac: Factory[B, M]
+    ): Either[C[E], M] = {
+      val bs = fac.newBuilder
+      val eb = ef.newBuilder
+
+      while (it.hasNext) {
```

Comment:
maybe it would make sense to introduce some flag (similar to `wasNone` from Option) to avoid collecting rights when there already exist left

Reply:
good idea 👍 also, when first error appears, we can clear `bs` builder immediately

Reply:
✅ done